File: src/main/java/com/GoldenMine/parser/pronoun/IPronoun.java
package com.GoldenMine.parser.pronoun;

import com.GoldenMine.parser.Sentence;
import com.GoldenMine.parser.Variable;
import com.GoldenMine.parser.VariableStorage;

public interface IPronoun {
    String getType();
    Variable perform(Sentence sentence, VariableStorage storage);
}

File: src/main/java/kr/goldenmine/parser/pronoun/IPronoun.kt
package kr.goldenmine.parser.pronoun

import kr.goldenmine.parser.Sentence
import kr.goldenmine.objects.Variable
import kr.goldenmine.parser.VariableStorage

interface IPronoun {
    val type: String
    fun perform(sentence: Sentence, metadata: List<Any>?, storage: VariableStorage): Variable?
}
